privly.safariextension/scripts/content_scritps/post_new_link.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @fileOverview Host Page Code
+ * This section is for managing the selection of form elements and posting
+ * of Privly-type links to those form elements.
+ *
+ * The flow of events followed in the script:
+ * 1. When the user right clicks the page it fires the contextmenu event.
+ *    The context information is added using setContextMenuEventUserInfo
+ * 2. The script records the target node
+ * 3. If the user selects the context menu "Privly New Message",
+ *    posting_process.js sends a message to this script to freeze posting.
+ *    This prevents future right click events from changing the destination
+ *    node.
+ * 4. Finally, the background script will message this script the Privly URL
+ *    to drop into the original form element.
+ */
+
+
+// Where the Privly URL will be placed is remembered by the contextmenu event
+var privlyUrlReceiptNode;
+
+// Variable used to indicate whether there is a pending operation
+var pendingPost = false;
+
+document.addEventListener("contextmenu", handleContextMenu, false);
+
+function handleContextMenu(evt) {
```

Comment:
For instance, `handleContextMenu` needs a function comment.

I think this file needs to be namespaced as well, although I would settle for a few more comments on the functions since you copied this from Chrome.